pkg/controller/operators/olm/operator.go

```diff
@@ -1211,6 +1235,114 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
+		result[operatorGroup.GetNamespace()] = NewNamespaceSet(operatorGroup.Status.Namespaces).IsAllNamespaces()
+	}
+	return result
+}
+
+func (a *Operator) syncOLMConfig(obj interface{}) (syncError error) {
+	a.logger.Info("Processing olmConfig")
+	olmConfig, ok := obj.(*v1.OLMConfig)
+	if !ok {
+		return fmt.Errorf("casting OLMConfig failed")
+	}
+
+	// Generate a map to track namespaces that are cluster scoped
+	operatorGroups, err := a.lister.OperatorsV1().OperatorGroupLister().List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	isNamespaceClusterScoped := isNamespaceClusterScopedMap(operatorGroups...)
+
+	// Get non-copied CSVs
+	requirement, err := labels.NewRequirement(v1alpha1.CopiedLabelKey, selection.DoesNotExist, []string{})
+	if err != nil {
+		return err
+	}
+
+	csvs, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().List(labels.NewSelector().Add(*requirement))
+	if err != nil {
+		return err
+	}
+
+	csvIsRequeued := false
+	for _, csv := range csvs {
+		// For each cluster scope installation
+		if !isNamespaceClusterScoped[csv.GetNamespace()] {
+			continue
+		}
+
+		// Get a count of the number of CSVS
+		requirement, err := labels.NewRequirement(v1alpha1.CopiedLabelKey, selection.Equals, []string{csv.GetNamespace()})
+		if err != nil {
+			return err
+		}
+
+		copiedCSVs, err := a.copiedCSVLister.List(labels.NewSelector().Add(*requirement))
+		if err != nil {
+			return err
+		}
+
+		// If the correct number of copied CSVs were found, continue
+		if len(copiedCSVs) == 0 == !olmConfig.CopiedCSVsAreEnabled() {
+			continue
+		}
+
+		// There were an incorrect number of copied CSVs, requeue the original CSV.
+		if err := a.csvQueueSet.Requeue(csv.GetNamespace(), csv.GetName()); err != nil {
+			return err
+		}
+		csvIsRequeued = true
+	}
+	if csvIsRequeued {
+		defer func() {
+			a.olmConfigQueue.AddAfter(olmConfig, time.Second*5)
+		}()
+	}
+
+	condition := getCopiedCSVsCondition(!olmConfig.CopiedCSVsAreEnabled(), csvIsRequeued)
+	// Update the olmConfig status if it has changed.
+	if !meta.IsStatusConditionPresentAndEqual(olmConfig.Status.Conditions, condition.Type, condition.Status) {
+		a.logger.Infof("Updating Condition: %v", condition)
+		meta.SetStatusCondition(&olmConfig.Status.Conditions, condition)
+		if _, err := a.client.OperatorsV1().OLMConfigs().UpdateStatus(context.TODO(), olmConfig, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getCopiedCSVsCondition(isDisabled, csvIsRequeued bool) metav1.Condition {
+	condition := metav1.Condition{
+		Type:               v1.DisabledCopiedCSVsConditionType,
+		LastTransitionTime: metav1.Now(),
+		Status:             metav1.ConditionFalse,
+	}
+	if !isDisabled {
+		condition.Reason = "CopiedCSVsEnabled"
+		condition.Message = "Copied CSVs are enabled"
+		if csvIsRequeued {
+			condition.Message = "No copied CSVs were founds for an operator installed in AllNamespace mode"
+		}
+	} else {
```

Comment:
nit: it looks like we could avoid the if/else conditional structure by returning the `condition` variable during the if clause and exiting early.

Reply:
Addressed in the latest version of the PR.